app/controllers/orders/client_summary.js

```diff
@@ -0,0 +1,35 @@
+import detail from "./detail";
+import Ember from 'ember';
+import AjaxPromise from 'stock/utils/ajax-promise';
+const { getOwner } = Ember;
+
+export default detail.extend({
+  showBeneficiaryModal: false,
+
+  actions: {
+    removeBeneficiaryModal() {
+      this.toggleProperty('showBeneficiaryModal');
+    },
+
+    deleteBeneficiary() {
```

Comment:
note : the `detail` base controller extends the `GoodcityController` which has a `deleteRecords`
You could replace a chunk of this code by `this.deleteRecords([ beneficiary ])`
It also handles the spinner (aka loadingView) 😄

Reply:
Thanks, I'll make changes :)

Reply:
Although that detail controller is for `orders` route and this is for `order`, I just noticed.

Reply:
Well, if you feel it's useful feel free to have your controller extend the `GoodcityController`, and feel free to add any handy/reusable code there as well : )